### ISSUE TITLE
DDRNet-23-slim, cannot reproduce results.

### ISSUE THREAD

**Issue by rawalkhirodkar** [chenjun2hao/DDRNet.pytorch#7]
Hello,
Thank you for the great work!

I am trying to reproduce the results of ddrnet-23_slim on Cityscapes val (77.83 mIoU) but could not do it.

- I tried training for more than 484 epochs - 600 epochs and  1500 epochs with 2, 5 and 8 GPUs.
- Evaluating with both heads model index 0 and model index 1
- The best performance is around 74.2 mIoU. 
- I also noticed, the results do not match when you use different number of gpus but train for the same number of epochs.

Here are the training logs,
[600epochs_5gpus_ddrnet23_slim_2021-06-02-20-21_train.txt](https://github.com/chenjun2hao/DDRNet.pytorch/files/6633671/600epochs_5gpus_ddrnet23_slim_2021-06-02-20-21_train.txt)
[600epochs_8gpu_ddrnet23_slim_2021-06-03-09-32_train.txt](https://github.com/chenjun2hao/DDRNet.pytorch/files/6633672/600epochs_8gpu_ddrnet23_slim_2021-06-03-09-32_train.txt)
[1500epochs_2gus_ddrnet23_slim_2021-06-04-18-36_train.txt](https://github.com/chenjun2hao/DDRNet.pytorch/files/6633688/1500epochs_2gus_ddrnet23_slim_2021-06-04-18-36_train.txt)

I would really appreciate your help. Thank you.

**Comment by ShihuaHuang95** [comment]
@rawalkhirodkar I got 75.6, but the performance is still worse than the stated.

**Comment by ShihuaHuang95** [comment]
Besides, the reproduced result of DDRNet-23 is low.

**Comment by chenjun2hao** [comment]
@rawalkhirodkar @ShihuaHuang95 ，i just train with 2*3080 gpus with the config files. i use the official imagenet pretrained model. Nothing else has been changed.

**Comment by rawalkhirodkar** [comment]
I think this has to do with the learning rate updates.
num_iters_per_epoch is a function of number of gpus which in turn controls the learning rate decay.

Training for longer (1500 epochs) with 8 GPUs reproduces the OWN performance with FLIP mentioned in the README.

Thanks.

**Comment by YAwei666** [comment]
> I think this has to do with the learning rate updates. num_iters_per_epoch is a function of number of gpus which in turn controls the learning rate decay.
> 
> Training for longer (1500 epochs) with 8 GPUs reproduces the OWN performance with FLIP mentioned in the README.
> 
> Thanks.

Are you use the same config, when train 1500 epoch with 8 GPUs reproduces the OWN performance? or you have changed something? I would really appreciate your help. Thank you.

**Comment by chenjun2hao** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
